packages/taxdat/R/covariate_helpers.R

```diff
@@ -1,2002 +1,2057 @@
+    message("Using the rgeoboundaries national shapefile for this country.")
+  } else {
+    message("Using the gadm national shapefile for this country.")
+    boundary_sf <- geodata::gadm(country = iso_code, 
+                                 level = admin_level, 
+                                 path = tempdir()) 
+    
+    if (admin_level > 1) {
+      warning('The current admin level is set at ', admin_level)
+    }
+    
+    # Fix colnames for compatibility with rest of code
+    boundary_sf <- boundary_sf %>% 
+      sf::st_as_sf() 
+    boundary_sf <- boundary_sf %>%
+      magrittr::set_colnames(.,tolower(colnames(boundary_sf))) %>%
+      dplyr::mutate(name_0 = country,
+                    shapeID = paste0(gid_0, "-ADM", admin_level, "-", !!rlang::sym(paste0("gid_", admin_level))),
+                    shapeType = paste0("ADM", admin_level)) %>% 
+      dplyr::select(shapeName = !!rlang::sym(paste0("name_", admin_level)),
+                    shapeID,
+                    shapeType)
+    
+    sf::st_crs(boundary_sf) <- sf::st_crs(4326)
+  }
+  
+  sf::st_geometry(boundary_sf) <- "geom"
+  
+  boundary_sf <- boundary_sf %>% 
+    dplyr::rename(location_period_id = shapeID)
+  
+  # Fix geometry collections if any
+  boundary_sf <- fix_geomcollections(boundary_sf)
+  
+  # Combine into multipolygons
+  boundary_sf <- sf::st_cast(boundary_sf, "MULTIPOLYGON")
+  boundary_sf <- boundary_sf %>% 
+    dplyr::group_by(shapeName) %>% 
+    dplyr::summarise(location_period_id = stringr::str_c(location_period_id, 
+                                                         collapse = "_"),
+                     shapeType = shapeType[1])
+  
+  return(boundary_sf)
+}
+
+
+#' Get all country admin units
+#' Gets admin 1 to 3 units for given country
+#' 
+#' @param iso_code 
+#'
+#' @return
+#' @export
+#'
+get_multi_country_admin_units <- function(iso_code,
+                                          admin_levels = 0:2,
+                                          lps = NULL,
+                                          clip_to_adm0 = TRUE,
+                                          dbuser = Sys.getenv("USER"),
+                                          source = "database") {
+  
+  if (!(source %in% c("database", "api"))) {
+    stop("Source for output shapfiles must either be 'database' or 'api', found: ", source)
+  }
+  
+  if (iso_code == "testing") {
+    # If testing run return the same shapefiles as the ones on location periods
+    return(lps)
+  } 
+  
+  
+  if (source == "database") {
+    adm_sf <- try(get_country_admin_units_db(iso_code  = iso_code, 
+                                             admin_level = admin_levels,
+                                             dbuser = dbuser))
+    if (!inherits(adm_sf, "try-error")) {
+      return(adm_sf)
+    }
+  }
+  
+  if (clip_to_adm0 & !(0 %in% admin_levels)) {
+    stop("Admin level 0 needs to be included if clip_to_adm0 = TRUE.")
+  }
+  
```

Comment:
Even though we used get_country_admin_units to generate the shapes that were loaded to get_country_admin_units_db, I'm concerned that we could edit the complex conditions under get_country_admin_units and then forget that we are loading the already-pulled shapes from the covariate database. We should at least add some sort of warning that the shapes are being re-pulled from external APIs if this part is triggered.

Reply:
Good point. I will add a hash of the code of get_country_admin_units to check whether the scripts are consistent.